File: E_Triple_Operations.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    int t;  // number of test cases
    cin >> t;

    while (t--) {
        int l, r;
        cin >> l >> r;  // read l and r for each test case

        int ans = 0;

        double log_val = log2(l) / log2(3);
        ans += ceil(log_val);
        if (ans == log_val) {
            ans += 1;
        }
        ans += ans;

        for (int i = l + 1; i <= r; ++i) {
            double log_val = log2(i) / log2(3);
            if (ceil(log_val) == log_val) {
                ans += 1;
            }
            ans += ceil(log_val);
        }

        cout << ans << endl;
    }

    return 0;
}

File: E_Triple_Operations.py
from math import log2, ceil

precompute = []
precompute.append(0)
for i in range(1, 2_00_001):
    log = log2(i)/log2(3)
    log += precompute[-1]
    if ceil(log) == log:
        precompute.append(log+1)
    else:
        precompute.append(ceil(log))
# print(precompute[:11])
t = int(input())  # number of test cases

for _ in range(t):
    l, r = map(int, input().split())  # read l and r for each test case

    print(int(precompute[r] + precompute[l] - precompute[l-1] * 2))  # print the answer